src/System.IO.FileSystem.Watcher/src/System/IO/FileSystemWatcher.Linux.cs

```diff
@@ -476,7 +478,19 @@ private void ProcessEvents()
                         string expandedName = null;
                         WatchedDirectory associatedDirectoryEntry = null;
 
-                        if (nextEvent.wd != -1) // wd is -1 for events like IN_Q_OVERFLOW that aren't tied to a particular watch descriptor
+                        // Two events take priority: Overflow and SymLinks. An overflow event means that we can't trust 
+                        // our state without restarting since we missed events and some of those events could be a 
+                        // directory create, meaning we wouldn't have added the directory to the watch and would not provide
+                        // correct data to the caller. The other event, SymLinks, should be filtered out due to the FSW contract.
```

Comment:
Why is the comment discussing prioritizing symlink-related events when the code doesn't actually do anything related to them?